frontend/packages/console-shared/src/hooks/select-list.ts

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { OnSelect } from '@patternfly/react-table';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+
+const getAllSelectedRows = <R extends K8sResourceCommon>(data: R[], selectedRows: string[]) =>
+  data.filter((row) => selectedRows.includes(row.metadata.uid));
+
+export const useSelectList = <R extends K8sResourceCommon>(
+  data: R[],
+  visibleRows: string[],
+  onRowSelected: (rows: R[]) => void,
+): {
+  onSelect: OnSelect;
+  selectedRows: string[];
+  updateSelectedRows: (rows: R[]) => void;
+} => {
+  const [selectedRows, setSelectedRows] = React.useState<string[]>([]);
+
+  const onSelect = (_event, isSelected, rowIndex, rowData) => {
+    let uniqueUIDs = [];
+    if (rowIndex === -1) {
+      if (isSelected) {
+        uniqueUIDs = _.uniq([...visibleRows, ...selectedRows]);
+      } else {
+        uniqueUIDs = _.uniq(selectedRows.filter((uid) => !visibleRows.includes(uid)));
+      }
+    } else {
+      uniqueUIDs = _.xor(selectedRows, [rowData?.props?.id]);
+    }
+    setSelectedRows(uniqueUIDs);
+    onRowSelected(getAllSelectedRows(data, uniqueUIDs));
```

Comment:
I dont see a lot of value in having `getAllSelectedRows` as separate function

Just inlining works fine (and is more transparent IMO)

```suggestion
    onRowSelected(data.filter((datum) => selectedRows.includes(datum.metadata.uid)));
```